addons/purchase_requisition/static/src/js/purchase_order_line_compare_list_controller.js

```diff
@@ -0,0 +1,68 @@
+/** @odoo-module alias=purchase_requisition.PurchaseOrderLineCompareListController **/
+
+import ListController from 'web.ListController';
+
+const PurchaseOrderLineCompareListController = ListController.extend({
+
+    // -------------------------------------------------------------------------
+    // Public
+    // -------------------------------------------------------------------------
+
+    init: function (parent, model, renderer, params) {
+        this.context = renderer.state.getContext();
+        this.best_date_ids = this.context.params.best_date_ids || {};
+        this.best_price_ids = this.context.params.best_price_ids || {};
+        return this._super(...arguments);
+    },
+
+    // -------------------------------------------------------------------------
+    // Handlers
+    // -------------------------------------------------------------------------
+    /**
+     * @override
+     */
+    _onButtonClicked: function (ev) {
+        if (ev.data.attrs.class && ev.data.attrs.class.split(' ').includes('o_clear_qty_buttons')) {
+            ev.stopPropagation();
+            var self = this;
+            return this._callButtonAction(ev.data.attrs, ev.data.record).then(() => {
+                const context = this.model.localData[0] && this.model.localData[0].getContext() || {};
+                return self._rpc({
+                    model: "purchase.order",
+                    method: 'get_tender_best_date_and_price_lines',
+                    args: [self.context.active_id],
+                    context: context,
+                }).then((best_date_and_price) => {
+                    self.renderer.best_date_ids = best_date_and_price[0] || {};
+                    self.renderer.best_price_ids = best_date_and_price[1] || {};
+                    self.reload();
+                });
+            });
+        } else {
+            this._super.apply(this, arguments);
+        }
```

Comment:
Suggestions:
- Again, no need to `self`;
- As `ev.stopPropagation` and `_callButtonAction` are all the `_super` does, I think we can call `_super` anyway and do the condition just after that;
- Use `async`/`await` instead of nesting multiple `then` ;
- No need to split class (`includes` exists for string too and should work here like it's expected), also we can do `(ev.data.attrs.class || '').includes` instead of `ev.data.attrs.class and ev.data.attrs.class.includes` (same result but shorter);
- Since `get_tender_best_date_and_price_lines` always returns something, no need to define a fallback value;
- I think we can use `this.context.active_ids` instead of `[this.context.active_id]` (not sure because `active_ids` is a `Proxy` of an `Array` instead of a real `Array`, but I guess as long as the `Proxy` doing correctly its job, it should be good).  
```suggestion
    _onButtonClicked: async function (ev) {
        await this._super(...arguments);
        if ((ev.data.attrs.class || '').includes('o_clear_qty_buttons')) {
            const best_date_and_price = await this._rpc({
                model: 'purchase.order',
                method: 'get_tender_best_date_and_price_lines',
                args: [this.context.active_id],
                context: this.model.localData[0] && this.model.localData[0].getContext() || {},
            });
            this.renderer.best_date_ids = best_date_and_price[0] || [];
            this.renderer.best_price_ids = best_date_and_price[1] || [];
            this.reload();
        }
```

Reply:
Ah I understand why you call manually `_callButtonAction` instead of `_super`... It's because `_onButtonClicked` returns nothing.

I think the right thing to do is to call `_super` here and then, in the `ListController._onButtonClicked`, returns `_callButtonAction` (suggestion for overriding purpose @odoo/rd-framework-js , are you OK with that ?)

Reply:
@svs-odoo We don't really mind, actually, as that legacy code in /web is about to be dropped. It's a shame, but I'm afraid you'll have to re-write the JS of this PR already.